app/controllers/board_controller.rb

```diff
@@ -1,21 +1,8 @@
 class BoardController < ApplicationController
-  #before_action :authenticate_user!
+  before_action :authenticate_user!
 
   def index
-    arr = []
-    mentoring_tracks = MentoringTrack.all
-    user_exercise_tracks = User.find(1).mentoring_tracks
-    @todo = Todo.new
-
-    mentoring_tracks.each do |m_track|
-      m_track.track_instances.each do |ti|
-        if ti.mentor_id == User.find(1).id
-          arr.push(ti)
-        end
-      end
-    end
-
-    render json: { Your_exercises: user_exercise_tracks.to_json(:include => {:track_instances => {:include => {:section_interactions => {:include => :todos}}}}), Mentee_exercises: arr.to_json(:include => {:section_interactions => {:include => :todos}})}
+    @mentoring_tracks = TrackInstance.where(mentor_id: current_user.id)
```

Comment:
@sashank1995 Let's change this to params[:user_id] instead. Also, handle the case when params[:user_id] is blank.